src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1115,7 +1115,7 @@ private Type GetCompatibleGenericBaseClass(Type type, Type baseType)
                 if (propGenSpec.DefaultIgnoreCondition == JsonIgnoreCondition.Always)
                 {
                     ignoredMembers ??= new Dictionary<string, PropertyGenerationSpec>();
-                    ignoredMembers.Add(propGenSpec.ClrName, propGenSpec);
+                    ignoredMembers.TryAdd(propGenSpec.ClrName, propGenSpec);
```

Comment:
Should really we be ignoring the result of `TryAdd` if it returns false? Is there a possibility that `PropertyGenerationSpec` varies significantly between the base and derived properties and the order of traversal can substantially impact how source code is being generated? cc @layomia

Reply:
If for example we only want to take the spec for the base property into account, we might want to introduce so that only the base member is inserted into the dictionary (using `Add` instead of `TryAdd`).